### DISCUSSION TITLE
Email Notification when a New Event is Scheduled

### DISCUSSION THREAD

**Discussion by prager** [solspace/craft-calendar#253] [Questions & Answers] [answered]
I am curious about email notifications that would be sent when a new event is scheduled. 

For example, I am scheduling an event and since I would need technicians to set up workstations with network access, I need the Calendar to send an email notification to the IT Department. Also, I need this for the events already scheduled. 

I created a mailto: link with subject and body on the template, but that is not sufficient and also kind of lame. I would like the notification to include all event's features, like dates and times that are created with the submission of the form.

What would it take to add such a feature? I am a decent PHP programmer, but new to Craft CMS.

**Craft and Plugin Info:**

- Craft PRO 4.6.0
- Calendar 4.0.11

**Comment by seandelaney** [comment, marked as answer]
Hello @prager 

I think the easiest option is to setup a custom module in Craft CMS:
[How to Build a Module | Craft CMS Documentation | 4.x](https://craftcms.com/docs/4.x/extend/module-guide.html) 

Within your custom module, you can listen for events. In this case, you would listen for the Calendars `SaveElementEvent` event. The `SaveElementEvent` event will contain the Event details and whether it's new or existing. 

I would suggest using Queues/Jobs for performance, so you don't freeze up the Craft CMS Control Panel.

Within the job, you then make use of Craft CMS's built in mail service:
[Queue Jobs | Craft CMS Documentation | 4.x](https://craftcms.com/docs/4.x/extend/queue-jobs.html#writing-a-job)
[Email | Craft CMS Documentation | 4.x](https://craftcms.com/docs/4.x/mail.html) 

Non-tested of what it might look like:
```
Event::on(
    \Solspace\Calendar\Services\EventsService::class,
    \Solspace\Calendar\Services\EventsService::EVENT_AFTER_SAVE,
    function (\Solspace\Calendar\Events\SaveElementEvent $event) {
        $element = $event->getElement(); // your Event
        $isNew = $event->isNew(); // New or existing
                    
        // Create new job, add to queue
        // The Job sets up a new message and sends
    }
);
```
